### PR TITLE
Adding the map view

### PR DESCRIPTION
**Pull request by jespino**
Adding a map view of all the events. I also added an utility to geolocalize all
the locations and keep them in a cache. New cities will require new
geolocations, but other than that I think is working great.

Of course, filtering also works, we can decide if we want to change the markers to show in green for example the open CFPs, but it is not important because filtering is already there.

imagen

### CODE REVIEW

**Comment by scraly** [comment]
Ohh, I like it! ❤️
It's so good to see the project with a lot of new useful features.

Just a bug I found:
image

Is it possible to see clearly the:
- date of the event
- the name with the link
- the CFP badge (and others badges)
?

**Comment by jespino** [comment]
There are 2 problems, the first one is real state. There is no much space there to put things.
The second problem is the one that you hit there, there is some kind of extrange behavior the first time you open the tooltip not getting all the needed space, if you open it again, you will see it with more content.
The name has directly the link there, so it should be enough to go to the place. I can add the dates, but if i put begin and end date is going to consume more real state...

I going to look into the first-open bug, but I think we can merge it, and iterate to get a better solution, something like whenever you click one event it shows all the info directly there in the tooltip with a back arrow icon, or something like that. But I see that happening in a second iteration. not right away.

**Comment by jespino** [comment]
I fixed the popup problem. I think we can address the dates problem in a different PR.

**Comment by scraly** [comment]
OK, so in another PR it can also be useful to address the bug of displaying only one label: CFP or closed caption. it can be useful to display the both:
image

File: page/src/components/MapView/MapView.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      }
+      if (!acc[cur.location]) {
+        acc[cur.location] = [];
+      }
+      acc[cur.location].push(cur);
+      return acc;
+    }, {})
+  }, [events]);
+
+  console.log(eventsByLocation)
```
Comment (scraly) [inline page/src/components/MapView/MapView.jsx:31]: remove console.log

**Comment by scraly** [comment]
A little fix to display simply the date for an event.

To add in `MapView.jsx` file after `console.log(eventsByLocation)`:

```
const monthNames = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  const formatDate = dates => {
    const startDate = `${new Date(dates[0]).getDate()}-${monthNames[
      new Date(dates[0]).getMonth()
    ].slice(0, 3)}`;
    let endDate = '';
    if (dates.length > 1) {
      endDate = new Date(dates[1]).getDate();
      endDate = `${new Date(dates[1]).getDate()}-${monthNames[new Date(dates[1]).getMonth()].slice(
        0,
        3
      )}`;
    }
    if (endDate) {
      return (
        <span>
          {startDate} -&gt; {endDate}
        </span>
      );
    }
    return <span>{startDate}</span>;
  };
```

And edit a line.

Before:

```
              {e.hyperlink ? <a href={e.hyperlink} target='_blank'>{e.name}</a> : <b>{e.name}</b>}
```

After:

```
              {formatDate(e.date)} {e.hyperlink ? <a href={e.hyperlink} target='_blank'>{e.name}</a> : <b>{e.name}</b>}
```

The result look like this:
image

**Review by scraly — CHANGES_REQUESTED** [review]

File: page/src/components/MapView/MapView.jsx (hunk 1 above)
Comment (jespino) [reply page/src/components/MapView/MapView.jsx:31]: Oops

**Comment by jespino** [comment]
@scraly I applied the proposed changes

**Comment by scraly** [comment]
Thanks :)

I have one question, how `geolocations.json` is created?
By what?

**Comment by scraly** [comment]
By the way, `run.sh` script is used locally.
You should update GH action `.github/workflows/ghpages.yml` to add the execution of the `geoCodes.js` tool:
```
node geoCodes.js
```

**Comment by jespino** [comment]
The geolocation of cities is not a cheap operation, most of the APIs doesn't allow you to freely find the location of a city, not only that, the ones that allows it, takes some time to locate the place. So we get and cache that upfront to just query that information on the flight. The cities that you see pointed in the maps are located by latitude and longitude, not by city name. This is faster, and not only that, this is "cheaper" in the sense of not geting the geolocation of cities for every single client. Also because I want to keep that quick to generate I put that in the repository to not need to get again the geolocation data.

Maybe, to keep track in an easier way of the changes, we can store that in a pretty format, instead of the current "everything in one line" format.

**Comment by jespino** [comment]
> By the way, `run.sh` script is used locally. You should update GH action `.github/workflows/ghpages.yml` to add the execution of the `geoCodes.js` tool:
> 
> ```
> node geoCodes.js
> ```

Done

File: page/src/components/ListView/ListView.jsx [hunk 2]
```diff
@@ -4,8 +4,6 @@ import {ArrowRight} from 'lucide-react';
 import {filterEvents} from '../../utils';
 import {useCustomContext} from 'app.context';
 
-import EventDisplay from '../EventDisplay/EventDisplay';
```
Comment (scraly) [inline page/src/components/ListView/ListView.jsx:7]: Warning, if you remove this line, list view is no longer running.

**Review by scraly — CHANGES_REQUESTED** [review]

File: page/src/components/ListView/ListView.jsx (hunk 2 above)
Comment (jespino) [reply page/src/components/ListView/ListView.jsx:7]: Not really, EventDisplay is not used anywhere in that file, the IDE was warning me that I left that there accidentally before.

**Comment by scraly** [comment]
> The geolocation of cities is not a cheap operation, most of the APIs doesn't allow you to freely find the location of a city, not only that, the ones that allows it, takes some time to locate the place. So we get and cache that upfront to just query that information on the flight. The cities that you see pointed in the maps are located by latitude and longitude, not by city name. This is faster, and not only that, this is "cheaper" in the sense of not geting the geolocation of cities for every single client. Also because I want to keep that quick to generate I put that in the repository to not need to get again the geolocation data.
> 
> Maybe, to keep track in an easier way of the changes, we can store that in a pretty format, instead of the current "everything in one line" format.

Ok, so if new events are added with new cities, how to update this file? :)

**Comment by jespino** [comment]
> > The geolocation of cities is not a cheap operation, most of the APIs doesn't allow you to freely find the location of a city, not only that, the ones that allows it, takes some time to locate the place. So we get and cache that upfront to just query that information on the flight. The cities that you see pointed in the maps are located by latitude and longitude, not by city name. This is faster, and not only that, this is "cheaper" in the sense of not geting the geolocation of cities for every single client. Also because I want to keep that quick to generate I put that in the repository to not need to get again the geolocation data.
> > Maybe, to keep track in an easier way of the changes, we can store that in a pretty format, instead of the current "everything in one line" format.
> 
> Ok, so if new events are added with new cities, how to update this file? :)

You only have to run the new `geoCodes.js` script, and it will generate the file for you taking the existing cache as reference and only searching for the new places.

This way, is not that bad that the CI does that for the maybe 5 or 6 new places where the location is not established, and time to time, we can generate the file locally and commit it to the repo.

**Comment by scraly** [comment]
Thanks, I understand.
I am thinking about a way to run the script and save the updated file, in another PR.

**Pull request merged**